experimental/uefi/channel/src/tests.rs

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use super::*;
+
+const BODY_LEN_MULTIPLICATOR: usize = 5;
+const MOCK_LARGE_MESSAGE_LEN: usize = MAX_FRAME_BODY_SIZE * BODY_LEN_MULTIPLICATOR;
+
+#[test]
+fn test_message_fragmentation() {
+    let mock_body = {
+        let mut x: u8 = 0;
+        let filler = || {
+            x = x.overflowing_add(1).0;
```

Comment:
Why overflowing? Shouldn't it be a checked add?

edit: I understand now that it is meant to just create some variation in the message body, but IMO it distracts from the point of this test. Perhaps consider filling it with some constant instead, unless you think the variation actually adds value to this test.

Reply:
I'll add a comment for extra context. The TL;DR is that I think it makes sense to check that the message is reconstructed in the correct order. I actually used the variant of just filling it with a constant first, but then it means we basically just check that the length of the message and reconstructed message matches.